app/javascript/articles/components/SaveButton.jsx

```diff
@@ -3,9 +3,28 @@ import PropTypes from 'prop-types';
 import { articlePropTypes } from '../../src/components/common-prop-types';
 
 export class SaveButton extends Component {
+  constructor() {
+    super();
+    this.state =
+      {
+        collectionDropboxValue: 'Default Collection',
+        allCollectionTitles: []
+      }
+  }
+
   componentDidMount() {
+    fetch('http://localhost:3000/api/user_collections',  {credentials: 'include'})
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ allCollectionTitles: data.collections })
+      })
+      .catch(error => window.alert(`Opps! Something went wrong! ${error.message}`))
     const { isBookmarked } = this.props;
-    this.setState({ buttonText: isBookmarked ? 'SAVED' : 'SAVE' });
+    this.setState({ buttonText: isBookmarked ? 'SAVED' : 'SAVE!' });
+  }
+
+  updateDropboxState = collectionName => {
+    this.setState({collectionDropboxValue: collectionName});
```

Comment:
Need to look into why the button isn't changing to 'SAVED' on click. It may be because the "isBookmarked" variable isn't changing (not sure what was changed to make this happen).